# Generated by Django 5.2.3 on 2025-08-10 20:34

import api.models
import django.db.models.deletion
from django.db import migrations, models


class Migration(migrations.Migration):

    initial = True

    dependencies = [
    ]

    operations = [
        migrations.CreateModel(
            name='Error',
            fields=[
                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('md5', models.CharField(max_length=32, unique=True)),
                ('path', models.CharField(db_index=True, max_length=2048)),
                ('server_version', models.CharField(db_index=True, default=api.models.get_version, max_length=32)),
                ('traceback', models.TextField()),
                ('created_at', models.DateTimeField(auto_now_add=True, db_index=True)),
                ('issue_url', models.URLField(blank=True, null=True)),
                ('issue_solved', models.BooleanField(default=False)),
                ('payload', models.BinaryField(blank=True, null=True)),
                ('json_payload', models.JSONField(blank=True, null=True)),
                ('request_metadata', models.JSONField(blank=True, null=True)),
            ],
            options={
                'ordering': ['-created_at'],
            },
        ),
        migrations.CreateModel(
            name='ErrorOccurrence',
            fields=[
                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('timestamp', models.DateTimeField(auto_now_add=True, db_index=True)),
                ('signature', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to='api.error')),
            ],
        ),
    ]
